pkg/server/commands/reverseexpand/reverse_expand.go

```diff
@@ -312,14 +312,29 @@ func (c *ReverseExpandQuery) execute(
 			}
 			err = c.execute(ctx, r, resultChan, intersectionOrExclusionInPreviousEdges, resolutionMetadata)
 			if err != nil {
-				return err
+				var returnedErrors *multierror.Error
```

Comment:
I would recommend we normalize and simplify this code to avoid duplication. May I suggest as an alternative the following snippet?

```go
var errs *multierror.Error

func _, edge := range edges {
    ...
    switch {
    ...
    case graph.ComputedUsersetEdge:
        ...
        err = c.execute(...)
        if err != nil {
          errs = multierror.Append(errs, err)
          break // this bit!
       }
    }
}

if err := pool.Wait(); err != nil {
    errs = multierror.Append(errs, err)
}

if errs.ErrorOrNil() != nil {
    telemetry.TraceError(span, err)
    return errs
}

return nil
```

This way we can avoid the duplications with the span handling (e.g. `span.End()`) as well as the telemetry and pool duplicated logic.

Reply:
Will do.  The only caveat is that the break needs to be labeled as per https://go.dev/ref/spec#Break_statements